internal/integration_test/engine/adhoc_test.go

```diff
@@ -65,7 +65,7 @@ func runAllTests(t *testing.T, tests map[string]func(t *testing.T, r wazero.Runt
 		name := name   // pin
 		testf := testf // pin
 		t.Run(name, func(t *testing.T) {
-			t.Parallel()
+			//t.Parallel()
```

Comment:
with this enabled, the context-related tests hang on arm64. It doesn't fail on amd64 but it may be due to rosetta emulation adding a little bit of delay (I haven't tested on native amd64 yet)